esmvaltool/preprocessor/_derive.py

```diff
@@ -37,37 +44,48 @@ def get_required(short_name):
 
 def derive(cubes, short_name):
     """Derive variable `short_name`"""
+    # Do nothing if variable is already available
+    if short_name == cubes[0].var_name:
+        return cubes[0]
+
+    # Derive
     functions = {
         'lwp': calc_lwp,
```

Comment:
One thing about the interface: we will have a lot of derivations that are computed in the same way. In particular, I am thinking in computing speed module from the velocity components (sispeed from siu and siv, sfcWind form ua and va ...). Do yo think it will be easy to adapt for those cases? (Not asking you to do it, but probably I will have to do it on the near future)

Reply:
Maybe you can tell me? I don't know how those quantities are computed. But if they take a number of cubes from the same model (but with different variables) as input and produce a single cube as output, then I think that should be no problem. Adding some configurable parameters to the namelist should also be fairly straightforward.

Reply:
It is mostly about the metadata of the generated variables. The same example we discuss for the force
```python
sfcWind = (uas**2+ vas**2) ** 0.5
```
But the algorithm is the same for a lot of similar variables
```python
sispeed = (siu**2+ siv**2) ** 0.5
```
We can use the same derivation function for all of them, but we need to be able to pass the function the derived variable metadata attributes that are currently hardcoded in the examples we have

Reply:
This would mean moving part of the derivation algorithm to the namelist, which is something I would avoid. The derivation should be somewhat automatic and related to a well-defined recipe, with a 1:1 correspondence variable->recipe. This might generate some code duplication, like in the examples you mentioned, but it would allow to keep the namelist clean without adding too many parameters.

Reply:
We can avoid the code duplication by extracting the computation part to a common functio. O, doubt solved. Thanks!